### DISCUSSION TITLE
Multi-level nested GraphQL data fetching

### DISCUSSION THREAD

**Discussion by TanveshT** [graphql-java/graphql-java#2652] [Q&A]
## Question

For Example:

schema:
```graphql
type Book {
   bookId: String!
   content: Content
}

union Content: XContent | YContent

type XContent {
   xContentTerm
   body
}

type YContent {
  yContentTerm
}
```

Here while retrieving `body` for `XContent`, which has a separate data fetcher. will have only the source of the parent i.e `XContent` whereas the body needs the `bookId` to fetch it from the data base. Is there a way we can address this via fetching the root `Book` while retrieving `body`

**Comment by bbakerman** [comment]
You want to use the field selection in your data fetcher

https://www.graphql-java.com/documentation/field-selection

and or the `graphql.schema.DataFetchingEnvironment#getExecutionStepInfo` which forms  a parent child chain backwards and also gives you access to arguments

**Reply by TanveshT** [reply]
Thank you @bbakerman, I will evaluate this and get back to you.

I am aware of the field-selection approach but that is not suitable for this as it overloads the base fetcher with knowledge of the extensions. But will surely check `graphql.schema.DataFetchingEnvironment#getExecutionStepInfo`

**Reply by TanveshT** [reply]
@bbakerman After evaluating I found out that `ExecutionStepInfo` gives information regarding the parent as the name also suggests and not the values fetched for the parent, like in this case `bookId`'s value isn't accessing. If ExecutionStepInfo does give us values can you please point the member variable which may hold this information?

**Reply by bbakerman** [reply]
We dont track valiues per parent / child.  Why?  Well it would mean we kept all fetched data in memory.  And thats just not efficient.  

So while we track parent types and paths (via ExecutionStepInfo) we dont keep fetched values in memory.

So if you need parent values (inside a child fetcher) then you have to invent a mechanism yourself. 

See https://www.graphql-java.com/blog/deep-dive-data-fetcher-results

and local context  and `graphql.GraphQLContext` - you can create a map like local context that can "capture" fetched data by path or name say.
